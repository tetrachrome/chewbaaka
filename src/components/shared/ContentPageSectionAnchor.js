/**
 * ContentPageSectionAnchor.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 07, 2020
 * Updated  : Jul 07, 2020
 */

import React from 'react';

function ContentPageSectionTitleToAnchorId(title) {
  return title.toLowerCase().replace(/\s/g , "-");
};

function ContentPageSectionAnchor(props) {
  return (
    <div id={ContentPageSectionTitleToAnchorId(props.title)} />
  );
}

export {
  ContentPageSectionTitleToAnchorId
}

export default ContentPageSectionAnchor;
