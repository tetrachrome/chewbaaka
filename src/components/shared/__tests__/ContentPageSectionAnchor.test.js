/**
 * ContentPageSectionAnchor.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 07, 2020
 * Updated  : Jul 07, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import ContentPageSectionAnchor from '../ContentPageSectionAnchor';

import { ContentPageSectionTitleToAnchorId } from '../ContentPageSectionAnchor'

test('renders ContentPageSectionAnchor component', () => {
  const title = "Cheetah and Man";

  render(<ContentPageSectionAnchor title={title} />);
});
