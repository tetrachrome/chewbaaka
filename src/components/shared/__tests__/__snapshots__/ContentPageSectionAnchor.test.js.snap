// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageSectionAnchor component snapshot 1`] = `
<div
  data-testid="ContentPageSectionAnchorComponentTestId"
  id="cheetah-and-man"
/>
`;
